### PR TITLE
feat: Algolia authentication - remove fail-safe feature

### PR DESCRIPTION
**Pull request by peterstarling**
Follow up on https://trello.com/c/rd7Z4bX0/1331-enable-authentication-for-the-search

### CODE REVIEW

File: apps/asap-server/src/handlers/webhooks/fetch-by-code/fetch-by-code.ts [hunk 1]
```diff
@@ -31,16 +30,9 @@ export const fetchUserByCodeHandlerFactory = (
...
-    try {
-      apiKey = algoliaClient.generateSecuredApiKey(algoliaSearchApiKey, {
-        validUntil: Date.now() + 36060, // which is one minute over the TTL of the ID token
-      });
-    } catch (e) {
-      apiKey = null;
-      logger.error(e, 'Error during generating the key for Algolia search');
-    }
+    const apiKey = algoliaClient.generateSecuredApiKey(algoliaSearchApiKey, {
+      validUntil: Date.now() + 36060, // which is one minute over the TTL of the ID token
```
Comment (harlan-kohll) [inline apps/asap-server/src/handlers/webhooks/fetch-by-code/fetch-by-code.ts:34]: Think there was another comment about this - can we put the `36060` into a `const`?

Comment (peterstarling) [reply apps/asap-server/src/handlers/webhooks/fetch-by-code/fetch-by-code.ts:34]: we could but erm.. what difference does it make :D? i guess i can put it in the config although to be fair its not a config attribute you would ever need to change and if you do its still global. But yeah if thats what you guys want

File: apps/asap-server/src/config.ts [hunk 2]
```diff
@@ -44,3 +44,4 @@ export const algoliaAppId = ALGOLIA_APP_ID || 'LVYWOPQ0A9';
 export const algoliaSearchApiKey = ALGOLIA_SEARCH_API_KEY || '';
 export const algoliaResearchOutputIndex =
   ALGOLIA_RESEARCH_OUTPUT_INDEX || 'asap-hub_research_outputs_dev';
+export const algoliaApiKeyTtl = 36060;
```
Comment (kmaid) [inline apps/asap-server/src/config.ts:47]: casing 😅

Comment (peterstarling) [reply apps/asap-server/src/config.ts:47]: whats wrong with it in this var name?

Comment (kmaid) [reply apps/asap-server/src/config.ts:47]: ALGOLIA_API_KEY_TTL

Comment (kmaid) [reply apps/asap-server/src/config.ts:47]: meh I don't know what the convention is in the backend. Thats how I would normally expect to find a constant

**Review by kmaid — APPROVED** [review]

**Pull request merged**